WooCommerce/Classes/ViewRelated/Orders/ShipmentProvidersViewController.swift

```diff
@@ -240,6 +240,12 @@ extension ShipmentProvidersViewController: UITableViewDelegate {
         }
 
         delegate?.shipmentProviderList(self, didSelect: provider, groupName: groupName)
+
+        navigateBack()
+    }
+
+    private func navigateBack() {
+        navigationController?.popViewController(animated: true)
```

Comment:
Since there won't be multiple navigation events in this screen, it's okay to get rid of the `navigateBack()` function and just add the `navigationController?.popViewController(animated: true)` to line 244.

Reply:
Sure. I tend to encapsulate things and separate responsibilities as much as possible, maybe sometimes a bit too much?